SFS_time_domain/driving_function_imp_nfchoa.m

```diff
@@ -129,24 +132,38 @@
     end
 end
 
-% -------------------------------------------------------------------------
-%                      ___
-%                1     \
-% D(phi0,w) = -------  /__     Hm(w) e^(im(phi0-phi_src))
-%             2 pi r0  m=-M..M
-%
-% see Spors et al. (2011), eq.(4) and (5)
-%--------------------------------------------------------------------------
+if strcmp('2.5D',dimension)
+    % -------------------------------------------------------------------------
+    %                      ___
+    %                1     \
+    % D(phi0,w) = -------  /__     Hm(w) e^(im(phi0-phi_src))
+    %             2 pi r0  m=-M..M
+    %
+    % see Spors et al. (2011), eq.(4) and (5)
+    %--------------------------------------------------------------------------
+    
+    % apply modal weighting
+    wm = modal_weighting(order,conf);  % modal weighting function
+    Hm = bsxfun(@times,Hm,wm);
+    
+    % apply phase shift due to source angle
+    m = 0:order;
+    dm = bsxfun(@times,Hm,exp(-1i*m*phi_src));
+    
+    % Spatial IFFT
+    dm = [conj(dm(:,end:-1:2)), dm];  % append coefficients for negative m
+    d = inverse_cht(dm,nls);
+    d = 1/(2*pi*R)*real(d);
+
+elseif strcmp('3D',dimension)
 
```

Comment:
Could you add a comment with the actual function and a reference similar to the 2.5D case.